### PR TITLE
Update: user avatar hooked up to v2 API

### PR DESCRIPTION
**Pull request by DanDeMicco**
WIP, some snapshot tests broken

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

File: src/components/ContentSidebar/ContentSidebar.js [hunk 1]
```diff
@@ -150,6 +153,27 @@ class ContentSidebar extends PureComponent<Props, State> {
         });
     }
 
+    /**
+     * Gets the user avatar URL
+     *
+     * @param {string} id the user id
+     * @return the user avatar URL string for a given user with access token attached
+     */
+    async getAvatarUrl(id: string): Promise<?string> {
```
Comment (DanDeMicco) [inline src/components/ContentSidebar/ContentSidebar.js]: Should I move this to Users API?

Comment (priyajeet) [reply src/components/ContentSidebar/ContentSidebar.js]: Sure...

Comment (priyajeet) [reply src/components/ContentSidebar/ContentSidebar.js]: I guess then it becomes similar to File.getDownloadUrl()

File: src/api/Users.js [hunk 2]
```diff
@@ -22,6 +20,34 @@ class Users extends Base {
...
+            throw new Error('Missing user id');
+        }
+        return `${this.getUrl(id)}/avatar`;
+    }
+
+    /**
+     * Gets the user avatar URL
+     *
+     * @param {string} userId the user id
+     * @return the user avatar URL string for a given user with access token attached
```
Comment (priyajeet) [inline src/api/Users.js]: {type} missing in doc

File: src/components/ContentSidebar/ActivityFeed/Avatar.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+     *
+     * @param {string} avatarUrl the user avatar url
+     */
+    getAvatarUrlHandler(avatarUrl: ?string) {
+        this.setState({
+            avatarUrl
+        });
+    }
+
+    getAvatarUrl() {
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/Avatar.js]: jsdocs on functions

File: src/components/ContentSidebar/ActivityFeed/Avatar.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+class Avatar extends React.PureComponent<Props, State> {
+    state = {
+        avatarUrl: null
+    };
+
+    /**
+     * Success handler for getting avatar url
+     *
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/Avatar.js]: missing param

File: src/components/ContentSidebar/ActivityFeed/Avatar.js (hunk 3 above)
Comment (priyajeet) [reply src/components/ContentSidebar/ActivityFeed/Avatar.js]: Also, is this function needed? Can't you just have it in didMount till there is a use case for having it outside?

Comment (priyajeet) [reply src/components/ContentSidebar/ActivityFeed/Avatar.js]: This would be a good candidate for async rendering when its out in later React versions.

Comment (DanDeMicco) [reply src/components/ContentSidebar/ActivityFeed/Avatar.js]: Updated the docs

Is it really needed to be its own function? No, but it makes it more testable

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #312

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

**Pull request merged**